packages/integration_test/README.md

```diff
@@ -7,78 +7,91 @@ and native Android instrumentation testing.
+Create a `integration_test/` directory for your package. In this directory,
+create a `<name>_test.dart`, using the following as a starting point to make
+assertions.
 
 ```dart
 import 'package:flutter_test/flutter_test.dart';
 import 'package:integration_test/integration_test.dart';
 
 void main() {
   IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
   testWidgets("failing test example", (WidgetTester tester) async {
     expect(2 + 2, equals(5));
   });
 }
 ```
 
-## Test locations
-
-It is recommended to put integration_test tests in the `test/` folder of the app
-or package. For example apps, if the integration_test test references example
-app code, it should go in `example/test/`. It is also acceptable to put
-integration_test tests in `test_driver/` folder so that they're alongside the
-runner app (see below).
-
-## Using Flutter Driver to Run Tests
-
-`IntegrationTestWidgetsTestBinding` supports launching the on-device tests with
-`flutter drive`. Note that the tests don't use the `FlutterDriver` API, they
-use `testWidgets` instead.
+### Driver Entrypoint
 
-Put the a file named `<package_name>_integration_test.dart` in the app'
-`test_driver` directory:
+An accompanying driver script will be needed that can be shared across all
+integration tests. Create a `integration_test_driver.dart` in the `test_driver/`
+directory with the following contents:
 
 ```dart
-import 'dart:async';
-
 import 'package:integration_test/integration_test_driver.dart';
 
-Future<void> main() async => integrationDriver();
+Future<void> main() => integrationDriver();
 ```
 
-To run a example app test with Flutter driver:
+### Package Structure
+
+Your package should have a structure that looks like this:
 
-```sh
-cd example
-flutter drive test/<package_name>_integration.dart
+```
+lib/
+  ...
+integration_test/
+  foo_test.dart
+  bar_test.dart
+test/
+  # Other unit tests go here.
+test_driver/
+  integration_test_driver.dart
 ```
 
-To test plugin APIs using Flutter driver:
+[Example](https://github.com/flutter/plugins/tree/master/packages/integration_test/example)
+
+## Using Flutter Driver to Run Tests
+
+These tests can be launched with the `flutter drive` command.
+
+To run `integration_test/foo_test.dart`:
 
 ```sh
-cd example
-flutter drive --driver=test_driver/<package_name>_test.dart test/<package_name>_integration_test.dart
+flutter drive \
+  --driver=test_driver/integration_test_driver.dart \
+  --target=integration_test/foo_test.dart
 ```
 
-You can run tests on web in release or profile mode.
+### Web
+
+Make sure you have [enabled web support](https://flutter.dev/docs/get-started/web#set-up)
+then [download and run](https://flutter.dev/docs/cookbook/testing/integration/introduction#6b-web)
+the web driver in another process.
 
-First you need to make sure you have downloaded the driver for the browser.
+Use following command to execute the tests, passing either `--release` or
+`--profile` as debug mode is not supported.
```

Comment:
Debug mode is now supported, and in fact, it's the only way of getting stack traces in web now.

Reply:
(At least in `master` :P)

Reply:
Hmm, on master i get this error when I run the tests, am I doing something wrongly?

```
$ flutter drive   --driver=test_driver/integration_test_driver.dart   --target=integration_test/example_test.dart -d web-server
Starting application: integration_test/example_test.dart
Flutter Driver web does not support running in debug mode.

Use --profile mode for testing application performance.
Use --release mode for testing correctness (with assertions).
```

Reply:
Hmmm, odd, I think your `master` may be a little outdated; here's the change that removes that limitation:

https://github.com/flutter/flutter/pull/62577

Reply:
My apologies, there was a mistake in my setup. Updated, thanks!